File: SCTE65/src/java/other/org/himalay/si/SCTE65DescriptorFactory.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
// 
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.si ;

import java.io.ByteArrayInputStream;
import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.io.PrintStream;
import java.util.ArrayList;
import org.himalay.msgs.runtime.Factory;
import org.himalay.msgs.runtime.FactoryProduct;
import org.himalay.msgs.runtime.Created;
import org.himalay.msgs.runtime.*;

@Created(date = "Sun Sep 14 22:48:37 EDT 2014")
public class SCTE65DescriptorFactory implements Factory{ 
    public static final int caption_service_descriptor = 0x86;
    public static final int revision_detection_descriptor = 0x93;

public static boolean isDebug = true;
public static  SCTE65Descriptor lastParsedObject= null;
    public static SCTE65Descriptor createMsg(DataInputStream istream, IntegerHolder iHolder1) throws IOException
    {
int iRead=0;
        descriptor_header desc_header;
        {
        int retVal= 0;
         desc_header = new descriptor_header();
retVal += desc_header.read(istream);
iRead= retVal;
}
        SCTE65Descriptor retVal= createMsg( desc_header,  istream,  iHolder1);
        iHolder1.setValue(iHolder1.getValue()+iRead);
        return retVal;
    }
  
    public static SCTE65Descriptor createMsg( org.himalay.si.descriptor_header temp, DataInputStream dis, IntegerHolder iHolder) throws IOException
    {
        int iRead= 0;
        SCTE65Descriptor retVal= null;
        try{
        switch( temp.getMessageType() )        {
        case 0x86 :
        {
        retVal= new caption_service_descriptor ();
            retVal.setHeader(temp);
        // has no spread
                iRead+=retVal.readNoHeader(dis);
            break;
        }
        case 0x93 :
        {
        retVal= new revision_detection_descriptor ();
            retVal.setHeader(temp);
        // has no spread
                iRead+=retVal.readNoHeader(dis);
            break;
        }
                                        default :
        {
        retVal= new generic_descriptor ();
            retVal.setHeader(temp);
        // has no spread
                 iRead+=retVal.readNoHeader(dis);
            break;
        }
        } 
        }catch(IOException ioex)
        {
        if (isDebug)
        {
        lastParsedObject = retVal;
        }
        throw ioex;
        }
        iHolder.setValue(iRead);
        return retVal;
    }
    
                public static abstract class SCTE65Descriptor extends BinMessage implements FactoryProduct {
    public SCTE65Descriptor() {
        super();
        // TODO Auto-generated constructor stub
        }
        public abstract void setHeader(org.himalay.si.descriptor_header header);
    public abstract org.himalay.si.descriptor_header getHeader();
        
    }
}

// End of code

File: libsi/src/bp/SCTE65DescriptorFactory.cpp
//@Generated(date = "Mon Feb 01 22:29:56 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
#include "SCTE65DescriptorFactory.h"
#include "caption_service_descriptor_base.h"
#include "caption_service_descriptor.h"
#include "revision_detection_descriptor_base.h"
#include "revision_detection_descriptor.h"
#include "generic_descriptor.h"
// Namespace
namespace org {
namespace himalay {
namespace si {

using namespace ::org::himalay::msgs::runtime;


org::himalay::si::SCTE65DescriptorFactory::SCTE65Descriptor::SCTE65Descriptor() {};
org::himalay::si::SCTE65DescriptorFactory::SCTE65Descriptor::~SCTE65Descriptor() {};
bool SCTE65DescriptorFactory::isDebug = true;
/**
 *
 *SCTE65DescriptorFactory::SCTE65DescriptorPtr SCTE65DescriptorFactory::lastParsedObject= NULL;
 *
 */
org_himalay_si_SCTE65DescriptorFactory_SCTE65Descriptor__PTR_TYPE SCTE65DescriptorFactory::createMsg(DataInputStream& istream, int& iHolder) throw(IOException)
{
    int retVal= 0;
    //org_himalay_si_descriptor_header__PTR_TYPE temp;
    org_himalay_si_descriptor_header__PTR_TYPE desc_header ;
    org_himalay_si_descriptor_header__NEW( desc_header); ;
    //org_himalay_si_descriptor_header__NEW(temp) ;//new org.himalay.msgs.cpp.Complex@af215c18());
    retVal += desc_header->read(istream);


    org_himalay_si_SCTE65DescriptorFactory_SCTE65Descriptor__PTR_TYPE retObj= createMsg( desc_header,  istream,  iHolder);
    iHolder=iHolder+retVal;
    return retObj;
}

org_himalay_si_SCTE65DescriptorFactory_SCTE65Descriptor__PTR_TYPE SCTE65DescriptorFactory::createMsg( org_himalay_si_descriptor_header__PTR_TYPE temp, DataInputStream& istream, int& iHolder) throw(IOException)
{
    int retVal= 0;
    org_himalay_si_SCTE65DescriptorFactory_SCTE65Descriptor__PTR_TYPE retObj;
    try {
        switch( temp->getMessageType() )        {
        case 0x86 :
        {
            retObj= org_himalay_si_SCTE65DescriptorFactory_SCTE65Descriptor__PTR_TYPE((SCTE65DescriptorFactory::SCTE65Descriptor *) new caption_service_descriptor ());
            retObj->setHeader(temp);
            retVal+=retObj->readNoHeader(istream);
            break;
        }
        case 0x93 :
        {
            retObj= org_himalay_si_SCTE65DescriptorFactory_SCTE65Descriptor__PTR_TYPE((SCTE65DescriptorFactory::SCTE65Descriptor *) new revision_detection_descriptor ());
            retObj->setHeader(temp);
            retVal+=retObj->readNoHeader(istream);
            break;
        }
        default :
        {
            retObj= org_himalay_si_SCTE65DescriptorFactory_SCTE65Descriptor__PTR_TYPE( (SCTE65DescriptorFactory::SCTE65Descriptor *) new generic_descriptor ());
            retObj->setHeader(temp);
            retVal+=retObj->readNoHeader(istream);
            break;
        }
        }
    } catch(IOException& ioex)
    {
        if (isDebug)
        {
//        lastParsedObject = retVal;
        }
        throw ioex;
    }
    iHolder=retVal;
    return retObj;
}
void intrusive_ptr_add_ref(org::himalay::si::SCTE65DescriptorFactory::SCTE65Descriptor* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::si::SCTE65DescriptorFactory::SCTE65Descriptor* p) {
    if (p->decreaseRef() == 0u) delete p;
};

}
}
} // End Namespace org.himalay.si




// End of code